### PR TITLE
fix package filter by replacing a JOIN with a subquery

### PR DESCRIPTION
**Pull request by velrest**
previously the filter returned too many packages because of a JOIN

### CODE REVIEW

File: timed/subscription/tests/test_package.py [hunk 1]
```diff
@@ -18,11 +19,15 @@ def test_subscription_package_list(auth_client):
...
 def test_subscription_package_filter_customer(auth_client):
-    other_project = ProjectFactory.create()
-    PackageFactory.create(billing_type=other_project.billing_type)
-
-    my_project = ProjectFactory.create()
-    package = PackageFactory.create(billing_type=my_project.billing_type)
+    customer = CustomerFactory.create()
+    billing_type = BillingTypeFactory.create()
+    package = PackageFactory.create(billing_type=billing_type)
+    ProjectFactory.create(billing_type=billing_type, customer=customer)
```
Comment (sliverc) [inline timed/subscription/tests/test_package.py]: you could use bulk_create and assign 2 so two are created.

File: timed/subscription/tests/test_package.py [hunk 2]
```diff
@@ -18,11 +19,15 @@ def test_subscription_package_list(auth_client):
...
+
+    my_project = ProjectFactory.create(
```
Comment (sliverc) [inline timed/subscription/tests/test_package.py]: my_project var is not needed as below you could simply adjust it to user create customer

File: timed/subscription/filters.py [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+
 from . import models
 
 
 class PackageFilter(FilterSet):
-    customer = NumberFilter(field_name='billing_type__projects__customer')
+    customer = NumberFilter(method='filter_customer')
+
+    def filter_customer(self, queryset, name, value):
+        billing_types = (Project.objects.filter(customer=value)
```
Comment (sliverc) [inline timed/subscription/filters.py]: what is the reason to use braces here?

**Comment by sliverc** [comment]
Ahh end you did branch this of current master it seems... needs rebasing ;)

File: timed/subscription/tests/test_package.py (hunk 2 above)
Comment (velrest) [reply timed/subscription/tests/test_package.py]: the second project is to check that only one package is returned.

File: timed/subscription/filters.py (hunk 3 above)
Comment (velrest) [reply timed/subscription/filters.py]: the line is too long

Comment (sliverc) [reply timed/subscription/filters.py]: I guess this is matter of taste... but something like the following is properly more readable:
```python
        billing_types = Project.objects.filter(
            customer=value
        ).values(
            'billing_type'
        )
```

**Pull request merged**